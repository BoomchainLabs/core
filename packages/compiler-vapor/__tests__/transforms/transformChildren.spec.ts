import { makeCompile } from './_utils'
import {
  transformChildren,
  transformElement,
  transformText,
  transformVIf,
} from '../../src'

const compileWithElementTransform = makeCompile({
  nodeTransforms: [
    transformText,
    transformVIf,
    transformElement,
    transformChildren,
  ],
})

describe('compiler: children transform', () => {
  test.todo('basic')

  test('children & sibling references', () => {
    const { code, helpers } = compileWithElementTransform(
      `<div>
        <p>{{ first }}</p>
        {{ second }}
        {{ third }}
        <p>{{ forth }}</p>
      </div>`,
    )
    expect(code).toMatchSnapshot()
    expect(Array.from(helpers)).containSubset([
      'next',
      'setText',
      'createTextNode',
      'insert',
      'template',
    ])
  })
})
