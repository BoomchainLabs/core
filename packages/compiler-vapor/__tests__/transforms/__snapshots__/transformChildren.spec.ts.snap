// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compiler: children transform > children & sibling references 1`] = `
"import { next as _next, createTextNode as _createTextNode, insert as _insert, setText as _setText, renderEffect as _renderEffect, template as _template } from 'vue';
const t0 = _template("<div><p></p> <!><p></p></div>")

export function render(_ctx) {
  const n4 = t0()
  const n0 = n4.firstChild
  const n3 = _next(n0, 2)
  const n2 = n3.nextSibling
  const n1 = _createTextNode(() => [_ctx.second, " ", _ctx.third, " "])
  _insert(n1, n4, n3)
  let _first, __first, _forth, __forth
  _renderEffect(() => {
    __first = _ctx.first
    if(_first !== __first) {
      _setText(n0, __first)
      _first = __first
    }
    __forth = _ctx.forth
    if(_forth !== __forth) {
      _setText(n2, __forth)
      _forth = __forth
    }
  })
  return n4
}"
`;
